sdk/eventhub/event-hubs/consuming-events.md

```diff
@@ -0,0 +1,85 @@
+# Consuming events
+
+The `EventHubConsumerClient` provides a _push-based_ model for receiving events.
+It exposes a `subscribe` method that lets you specify:
+
+- Whether to read from a specific partition or all partitions.
+- User-provided functions that will be called while your subscription is active.
+- `SubscribeOptions` that let you control how events are received, including batching options.
+
+`subscribe` also returns a `Subscription` object that you can use to stop receiving events or
+to determine if the subscription is actively receiving events.
+
+Calling `subscribe` immediately starts the subscription.
+Once the subscription starts, the following things happen for **each** partition being read.
+
+1.  If supplied, the `partitionInitialize` function is invoked each time the subscription is
+    about to begin reading from a partition.
```

Comment:
We can use a line break here before talking about what is passed to the function.
```suggestion
    about to begin reading from a partition.
    
```